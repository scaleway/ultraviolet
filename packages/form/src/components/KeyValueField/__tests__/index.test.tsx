import { screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { renderWithForm } from '@utils/test'
import { describe, expect, it } from 'vitest'
import { KeyValueField } from '..'

describe('KeyValueField', () => {
  it('should render with default props', async () => {
    const { asFragment } = renderWithForm(
      <KeyValueField
        addButton={{
          maxSizeReachedTooltip:
            'This is a tooltip when the max size is reached',
          name: 'add',
          tooltip: 'This is a tooltip',
        }}
        inputKey={{
          label: 'key',
        }}
        inputValue={{
          label: 'value',
        }}
        name="test"
      />,
    )
    const addButton = screen.getByTestId('add-button')
    await userEvent.click(addButton)

    const removeButton = screen.getByTestId('remove-button-0')
    await userEvent.click(removeButton)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render with default props & max size', () => {
    const { asFragment } = renderWithForm(
      <KeyValueField
        addButton={{
          maxSizeReachedTooltip:
            'This is a tooltip when the max size is reached',
          name: 'add',
          tooltip: 'This is a tooltip',
        }}
        inputKey={{
          label: 'key',
        }}
        inputValue={{
          label: 'value',
        }}
        maxSize={42}
        name="test"
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render with default props in readonly mode', () => {
    const { asFragment } = renderWithForm(
      <KeyValueField
        addButton={{
          maxSizeReachedTooltip:
            'This is a tooltip when the max size is reached',
          name: 'add',
          tooltip: 'This is a tooltip',
        }}
        inputKey={{
          label: 'key',
        }}
        inputValue={{
          label: 'value',
        }}
        name="test"
        readOnly
      />,
    )
    const addButton = screen.getByTestId('add-button')
    expect(addButton).toBeDisabled()
    expect(asFragment()).toMatchSnapshot()
  })
})
