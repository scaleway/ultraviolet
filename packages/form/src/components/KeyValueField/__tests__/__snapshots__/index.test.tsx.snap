// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`keyValueField > should render with default props & max size 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.5rem;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-4 {
  vertical-align: middle;
  fill: currentColor;
  height: 1rem;
  width: 1rem;
  min-width: 1rem;
  min-height: 1rem;
}

.emotion-4 .fillStroke {
  stroke: currentColor;
  fill: none;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        class="emotion-0 emotion-1"
      >
        <div
          class="emotion-2 emotion-1"
        >
          <div
            aria-controls="«r8»"
            aria-describedby="«r8»"
            class="uv_w40vpo9"
            tabindex="0"
          >
            <button
              class="uv_e1wcoe0 uv_e1wcoe1 uv_e1wcoe6 uv_e1wcoek uv_e1wcoeh uv_e1wcoe7 uv_e1wcoeu"
              data-testid="add-button"
              type="button"
            >
              <svg
                class="emotion-4 emotion-5"
                viewBox="0 0 16 16"
              >
                <path
                  d="M8.75 3.75a.75.75 0 0 0-1.5 0v3.5h-3.5a.75.75 0 0 0 0 1.5h3.5v3.5a.75.75 0 0 0 1.5 0v-3.5h3.5a.75.75 0 0 0 0-1.5h-3.5z"
                />
              </svg>
              add
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`keyValueField > should render with default props 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.5rem;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-4 {
  vertical-align: middle;
  fill: currentColor;
  height: 1rem;
  width: 1rem;
  min-width: 1rem;
  min-height: 1rem;
}

.emotion-4 .fillStroke {
  stroke: currentColor;
  fill: none;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        class="emotion-0 emotion-1"
      >
        <div
          class="emotion-2 emotion-1"
        >
          <div
            aria-controls="«r0»"
            aria-describedby="«r0»"
            class="uv_w40vpo9"
            tabindex="0"
          >
            <button
              class="uv_e1wcoe0 uv_e1wcoe1 uv_e1wcoe6 uv_e1wcoek uv_e1wcoeh uv_e1wcoe7 uv_e1wcoeu"
              data-testid="add-button"
              type="button"
            >
              <svg
                class="emotion-4 emotion-5"
                viewBox="0 0 16 16"
              >
                <path
                  d="M8.75 3.75a.75.75 0 0 0-1.5 0v3.5h-3.5a.75.75 0 0 0 0 1.5h3.5v3.5a.75.75 0 0 0 1.5 0v-3.5h3.5a.75.75 0 0 0 0-1.5h-3.5z"
                />
              </svg>
              add
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`keyValueField > should render with default props in readonly mode 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.5rem;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-4 {
  vertical-align: middle;
  fill: currentColor;
  height: 1rem;
  width: 1rem;
  min-width: 1rem;
  min-height: 1rem;
}

.emotion-4 .fillStroke {
  stroke: currentColor;
  fill: none;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        class="emotion-0 emotion-1"
      >
        <div
          class="emotion-2 emotion-1"
        >
          <div
            aria-controls="«r9»"
            aria-describedby="«r9»"
            class="uv_w40vpo9"
            tabindex="0"
          >
            <button
              class="uv_e1wcoe0 uv_e1wcoe1 uv_e1wcoe6 uv_e1wcoej uv_e1wcoeh uv_e1wcoe7 uv_e1wcoeu"
              data-testid="add-button"
              disabled=""
              type="button"
            >
              <svg
                class="emotion-4 emotion-5"
                viewBox="0 0 16 16"
              >
                <path
                  d="M8.75 3.75a.75.75 0 0 0-1.5 0v3.5h-3.5a.75.75 0 0 0 0 1.5h3.5v3.5a.75.75 0 0 0 1.5 0v-3.5h3.5a.75.75 0 0 0 0-1.5h-3.5z"
                />
              </svg>
              add
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
